Add ProVerif model for basic hash (authentication)

nadimkobeissi opened the pull request:
This pull request adds a proposed ProVerif model to demonstrate basic hash functionality. The model comes with pedagogical comments intended to help students.

nadimkobeissi committed:
Add ProVerif model for basic hash (authentication)

This pull request adds a proposed ProVerif model to demonstrate basic hash functionality. The model comes with pedagogical comments intended to help students.

charlie-j commented on README.md:
```diff
@@ -69,9 +69,9 @@ subfolder: [basic-hash](basic-hash/)
-| Proverif    | Active   |    ∞     |   ∞    | None        | Symbolic       | Hash       | Auth, ⚡RA | Easy (Auth,RA), Hard (Unlo)  |  WIP   |
-| DY*         | Active   |    ∞     |   ∞    | Yes         | Symbolic       | MAC        | Auth       | Easy (Auth)                 |  ✅    |
+| Tamarin     | Active   |    ∞     |   ∞    | None        | Symbolic       | Hash       | Auth, ⚡RA  | Easy (Auth,RA), Hard (Unli) |   ✅   |
+| Proverif    | Active   |    ∞     |   ∞    | None        | Symbolic       | Hash       | Auth       | Easy                        |   ✅   |
```
We can have the Replay Attack mark easily (see next comment below), and I still think Unlinkability would be hard to get, I would leave it in the difficulty appreciation.

charlie-j commented on basic-hash/ProVerif/basic-hash-auth.pv:
```diff
@@ -0,0 +1,76 @@
+(*
+	Basic Hash Authentication model based on CryptoVerif equivalent found in D. Baelde, S. Delaune, A. Koutsos, C. Jacomme, and S. Moreau,
```
Sorry, you are a bit early, we are still ironing out the contributing guidelines. But so, I know it is not yet the case for the other basic-hash example, but as mentioned in the README, we want all files to have in the header a with common set of informations, synced with the README line, see for instance https://github.com/proof-ladders/protocol-ladder/blob/0f54196550a31d57489111692803e1eccabc0555/signedDH/signedDH.pv#L2

charlie-j commented on basic-hash/ProVerif/basic-hash-auth.pv:
```diff
@@ -0,0 +1,76 @@
+(* We define queries to check the capabilities of the attacker. *)
+(* Essentially, this query mandates that if a session is accepted by the receiver with a certain nonce and hash, *)
+(* then the session must have started with the same nonce and hash. *)
+query xn: nonce, xh:hashed; event(acceptR(xn,xh)) ==> event(startT(xn,xh)).
```
can you add the inj variant?
`query xn: nonce, xh:hashed; inj-event(acceptR(xn,xh)) ==> inj-event(startT(xn,xh)).` 

It instantly allows to capture the replay attack.

charlie-j reviewed (commented):
Thanks! 

I just made some minor comments, otherwise the file looks good :)

nadimkobeissi committed:
Improve basic-hash-auth-replay.pv

- Add inj-event query
- Use correct format according to contribution guidelines

nadimkobeissi committed:
Improve README.md

- Fix many formatting errors
- Add Replay property to ProVerif basic-hash model

nadimkobeissi committed:
Merge branch 'main' into main

nadimkobeissi committed:
Further README.md improvement

nadimkobeissi commented:
@charlie-j Comments addressed!

nadimkobeissi committed:
Merge branch 'main' into main

nadimkobeissi commented:
@charlie-j Wondering if there are any updates here?

charlie-j committed:
Merge branch 'main' into main

charlie-j commented:
Here  you go, thanks!
